### PR TITLE
Port Controller and Logger to C++

### PR DESCRIPTION
**Pull request by EricPickup**
### What are you trying to do?

Port the Controller and Logger classes over to C++ code.

##### For Controller:

virtualcar imports the controller class in the main program file (`virtualcar.c`) using `#include "include/nodes/controller.c"`. Updating it to C++, this file is now `controller.cpp`. You shouldn't include CPP files into other CPP files. I created a header file `controller.h` which our equivalent of `virtualcar.c` will include.

##### For Logger:

Basically porting over writing to files, no major changes

### CODE REVIEW

**Review by DSchana — CHANGES_REQUESTED** [review]
Update port-controller with the merge from yesterday. Also, `controller.c` doesn't implement from any headers, so you can just change the name to `controller.h` and not need the .cpp file for it.
